packages/clerk-js/src/ui/utils/createCustomPages.tsx

```diff
@@ -0,0 +1,233 @@
+};
+
+// Set the path of the first route to '/' or if the first route is account or security, set the path of both account and security to '/'
+const setFirstPathToRoot = (routes: NavbarRoute[]) => {
+  if (routes[0].id === 'account' || routes[0].id === 'security') {
+    return routes.map(r => {
+      if (r.id === 'account' || r.id === 'security') {
+        return { ...r, path: '/' };
+      }
+      return r;
+    });
+  } else {
+    return routes.map((r, index) => (index === 0 ? { ...r, path: '/' } : r));
+  }
+};
+
+const checkForDuplicateUsageOfReorderingItems = (customPages: CustomPage[]) => {
+  const reorderItems = customPages.filter(cp => isAccountReorderItem(cp) || isSecurityReorderItem(cp));
+  reorderItems.reduce((acc, cp) => {
+    if (acc.includes(cp.label)) {
+      console.error(
+        `The "${cp.label}" item is used more than once when reordering UserProfile pages. This may cause unexpected behavior.`,
+      );
+    }
+    return [...acc, cp.label];
+  }, [] as string[]);
+};
+
+const warnForDuplicatePaths = (routes: NavbarRoute[]) => {
+  const paths = routes
+    .filter(({ external, path }) => !external && path !== '/' && path !== 'account')
+    .map(({ path }) => path);
+  const duplicatePaths = paths.filter((p, index) => paths.indexOf(p) !== index);
+  duplicatePaths.forEach(p => {
+    console.error(`Duplicate path "${p}" found in custom pages. This may cause unexpected behavior.`);
+  });
+};
+
+const isValidPageItem = (cp: CustomPage): cp is CustomPage => {
+  return isCustomPage(cp) || isCustomLink(cp) || isAccountReorderItem(cp) || isSecurityReorderItem(cp);
+};
+
+const isCustomPage = (cp: CustomPage): cp is UserProfileCustomPage => {
+  return !!cp.url && !!cp.label && !!cp.mount && !!cp.unmount && !!cp.mountIcon && !!cp.unmountIcon;
+};
+
+const isCustomLink = (cp: CustomPage): cp is UserProfileCustomLink => {
+  return !!cp.url && !!cp.label && !cp.mount && !cp.unmount && !!cp.mountIcon && !!cp.unmountIcon;
+};
+
+const isAccountReorderItem = (cp: CustomPage): cp is UserProfileReorderItem => {
+  return !cp.url && !cp.mount && !cp.unmount && !cp.mountIcon && !cp.unmountIcon && cp.label === 'account';
+};
+
+const isSecurityReorderItem = (cp: CustomPage): cp is UserProfileReorderItem => {
+  return !cp.url && !cp.mount && !cp.unmount && !cp.mountIcon && !cp.unmountIcon && cp.label === 'security';
+};
+
+const sanitizeCustomPageURL = (url: string): string => {
+  if (!url) {
+    throw new Error('URL is required for custom pages');
+  }
+  if (isValidUrl(url)) {
+    throw new Error('Absolute URLs are not supported for custom pages');
+  }
+  return (url as string).charAt(0) === '/' && (url as string).length > 1 ? (url as string).substring(1) : url;
+};
+
+const sanitizeCustomLinkURL = (url: string): string => {
+  if (!url) {
+    throw new Error('URL is required for custom links');
+  }
+  if (isValidUrl(url)) {
+    return url;
+  }
+  return (url as string).charAt(0) === '/' ? url : `/${url}`;
+};
```

Comment:
We want the Custom Link items (external links) to have a leading `/` so that navigate works correctly, and on the other hand Custom Page items need a relative segment, to navigate inside the UserProfile/OrganizationProfile routing mechanism.